enum.js

```diff
@@ -1,7 +1,16 @@
 const cleanroom = Object.freeze(Object.create(null))
 
-const prototype = {
-  unwrap () { return this.value },
+class Enum {
+  constructor (value, members, label) {
+    Object.defineProperties(this, {
+      value: { value: value, enumerable: true },
+      members: { value: members, enumerable: false },
+      label: { value: label, enumerable: true },
+    })
```

Comment:
So all we're doing here is setting each incoming argument to be an object of itself plus whether it is enumerable? Do we use the enumerable boolean anywhere?

Reply:
It's a little different than that. What I wrote here is exactly equivalent to this:

```es6
constructor (value, members, label) {
  this.value = value
  this.members = members
  Object.defineProperty(this, 'members', { enumerable: false })
  this.label = label

  Object.freeze(this)
}
```

I just want the "members" property to not be enumerable so debug tools and pretty printers don't display it.

See: [Object.defineProperties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperties)